test/e2e/framework/metrics/metrics_grabber.go

```diff
@@ -334,6 +344,12 @@ func (g *Grabber) Grab(ctx context.Context) (Collection, error) {
 		} else {
 			result.APIServerMetrics = metrics
 		}
+		metrics, err = g.GrabMetricsSLIsFromAPIServer(ctx)
+		if err != nil {
+			errs = append(errs, err)
```

Comment:
so tests will fail when tested with the feature gate turned off? It looks like a common pattern, so this is not the comment to this PR. Just in general, not an ideal practice

/lgtm

Reply:
feature gate should be set to True for beta features, which is what we are doing in this PR and hence have added this test. But if the feature was still alpha, we wouldn't have a test for it.   @logicalhan to confirm this.

Reply:
Yeah, there isn't an API we can hit in order to determine feature enablement, so we can't actually toggle tests depending on enablement. Therefore we end up with this pattern.